src/Main.cc

```diff
@@ -216,8 +242,8 @@ int main(int argc, char* argv[])
 #ifdef BAZEL_CURRENT_REPOSITORY
   using rules_cc::cc::runfiles::Runfiles;
   std::string error;
-  std::unique_ptr<Runfiles> runfiles(
-      Runfiles::Create(argv[0], BAZEL_CURRENT_REPOSITORY, &error));
+  std::unique_ptr<Runfiles> runfiles(Runfiles::Create(
+      getProgramLocation().string(), BAZEL_CURRENT_REPOSITORY, &error));
```

Comment:
does this comile ? getProgramLocation() returns an optional.

I'd expect something along the lines of
```c++
 auto maybe_location = getProgramLocation();
 Runfiles::Create(maybe_location.has_value() ? *maybe_location : std::string(), ...);
```

Reply:
alternatively, if optinal is not really used, just modify the `getProgramLocation()` to return a string and make it an empty string in case it can't be determined.

Reply:
Fixed compile error.

Rather than add iffy code "log and continue", just throw an exception if the location can't be gotten: early failure.

Reply:
sounds good.